bsp/src/org/jetbrains/bsp/project/resolver/BspResolverLogic.scala

```diff
@@ -318,31 +307,42 @@ private[resolver] object BspResolverLogic {
     data
   }
 
+  private[resolver] def sharedModuleId(targets: Seq[BuildTarget]): String = {
```

Comment:
can you add an explanation or better unit test for what this function does? In particular the regex